pi/__init__.py

```diff
@@ -91,3 +92,99 @@ def flags(*args, **kwargs):
 
 
 FloatTensor = tensor
+
+import builtins
+
+builtin_int = builtins.int
+builtin_bool = builtins.bool
+builtin_float = builtins.float
+
+
+class pi_bool_meta(type):
+    def __instancecheck__(self, other):
+        return isinstance(other, builtin_bool)
+
+    def __call__(self, x, base=None):
+        if isinstance(x, Tensor):
+            return ops.Bool(x)
+        elif isinstance(x, Torch_BoolValue):
+            try:
+                return builtin_bool(x)
+            except:
+                return x
+        else:
+            return builtin_bool(x)
+
+
+class pi_bool(builtin_int, metaclass=pi_bool_meta):
```

Comment:
you can't subclass `builtins.bool` because it's `@final` but `bool` is a subclass of `int` anyway so just do the same thing.